client/faucet/src/main.rs

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_faucet::mint;
+use libra_logger::prelude::info;
+use std::fmt;
+use structopt::StructOpt;
+use warp::Filter;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "Libra Faucet",
+    author = "The Libra Association",
+    about = "Libra Testnet utitlty service for creating test account and minting test coins"
+)]
+struct Args {
+    /// Faucet service listen address
+    #[structopt(short = "a", long, default_value = "127.0.0.1")]
+    pub address: String,
+    /// Faucet service listen port
+    #[structopt(short = "p", long, default_value = "80")]
+    pub port: u16,
+    /// Libra fullnode/validator server URL
+    #[structopt(short = "s", long, default_value = "https://testnet.libra.org/v1")]
+    pub server_url: String,
+    /// Path to the private key for creating test account and minting coins.
+    /// To keep Testnet simple, we used one private key for both treasury compliance account and testnet
+    /// designated dealer account, hence here we only accept one private key.
+    /// To manually generate a keypair, use generate-key:
+    /// `cargo run -p generate-keypair -- -o <output_file_path>`
+    #[structopt(short = "m", long, default_value = "/opt/libra/etc/mint.key")]
+    pub mint_key_file_path: String,
+    /// Chain ID of the network this client is connecting to.
+    /// For mainnet: \"MAINNET\" or 1, testnet: \"TESTNET\" or 2, devnet: \"DEVNET\" or 3, \
+    /// local swarm: \"TESTING\" or 4
+    /// Note: Chain ID of 0 is not allowed; Use number if chain id is not predefined.
+    #[structopt(short = "c", long, default_value = "2")]
+    pub chain_id: libra_types::chain_id::ChainId,
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Args::from_args();
+    libra_logger::Logger::new().init();
+
+    let address: std::net::SocketAddr = format!("{}:{}", args.address, args.port)
+        .parse()
+        .expect("invalid address or port number");
+
+    info!(
+        "[faucet]: chain id: {}, server url: {}",
+        args.chain_id,
+        args.server_url.as_str(),
+    );
+    let service = std::sync::Arc::new(mint::Service::new(
+        args.server_url,
+        args.chain_id,
+        args.mint_key_file_path,
+    ));
+
+    info!("[faucet]: running on: {}", address);
+    warp::serve(routes(service)).run(address).await;
+}
+
+fn routes(
+    service: std::sync::Arc<mint::Service>,
+) -> warp::filters::BoxedFilter<(impl warp::Reply,)> {
+    // POST /mint?amount=25&auth_key=xxx&currency_code=XXX
+    let mint = warp::path("mint")
```

Comment:
Can you handle minting at `/` as well, old clients expect that path.

Reply:
you are right, I forgot this old path.